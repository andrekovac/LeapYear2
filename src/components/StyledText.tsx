import { Animated } from "react-native";
import styled from "styled-components/native";

type StyledTextProps = {
  size?: number;
  color?: string;
};

const StyledText = styled(Animated.Text)<StyledTextProps>`
  font-size: ${props => (props.size ? props.size + "px" : "30px")};
  font-weight: 200;
  text-align: center;
  color: ${props => props.color ?? "black"};
`;

export default StyledText;
