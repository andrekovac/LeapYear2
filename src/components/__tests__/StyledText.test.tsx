import React from "react";
import renderer from "react-test-renderer";

import StyledText from "../StyledText";

describe("StyledText", () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<StyledText />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly with size prop`, () => {
    const tree = renderer.create(<StyledText size={100} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly with color prop`, () => {
    const tree = renderer.create(<StyledText color="blue" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
