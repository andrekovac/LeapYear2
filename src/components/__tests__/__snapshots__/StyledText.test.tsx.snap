// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "black",
      "fontSize": 30,
      "fontWeight": "200",
      "textAlign": "center",
    }
  }
/>
`;

exports[`StyledText renders correctly with color prop 1`] = `
<Text
  color="blue"
  style={
    Object {
      "color": "blue",
      "fontSize": 30,
      "fontWeight": "200",
      "textAlign": "center",
    }
  }
/>
`;

exports[`StyledText renders correctly with size prop 1`] = `
<Text
  size={100}
  style={
    Object {
      "color": "black",
      "fontSize": 100,
      "fontWeight": "200",
      "textAlign": "center",
    }
  }
/>
`;
